app/services/statistics/base.rb

```diff
@@ -0,0 +1,29 @@
+module Statistics
+  class Base
+    attr_reader :search, :title, :count_by
+
+    def initialize(options = {})
+      @id       = options[:id]
+      @title    = options[:title]
+      @search   = options[:search]
+      @url      = options[:url]
+      @count_by = options[:count_by]
+    end
+
+    def calculate
+      raise NotImplementedError, "Method 'calculate' method needs to be implemented"
+    end
+
+    def id
+      @id || count_by.to_s
```

Comment:
some of the queries required string (e.g. facts) instead of changing that and everywhere that uses it, I chose to change it here.